django/db/backends/postgresql/operations.py

```diff
@@ -170,7 +169,7 @@ def sequence_reset_sql(self, style, model_list):
             # and column name (available since PostgreSQL 8)
 
             for f in model._meta.local_fields:
-                if isinstance(f, models.AutoField):
+                if f.uses_sequence:
```

Comment:
This a backward incompatible change that should be [deprecated](https://docs.djangoproject.com/en/stable/internals/contributing/writing-code/submitting-patches/#deprecating-a-feature) and proceed separately.

Reply:
That's not quite true. If `AutoField` has `uses_sequence = True` via inheritance from `AutoFieldMixin`, this will still work for all subclasses of `AutoField`.

However, there are many other cases of using `instance(…, AutoField)` that ought to be changed and they should be tackled together.

`uses_sequence` is a bad name for what we really want which is that the database is returning a generated value cf. generated columns, column defaults.

Reply:
I have my own reservations about the name `uses_sequence` and am open to suggestions for improvement there, but I didn't mean to use it to indicate that the DB could provide a value. Rather, it was specifically meant to indicate that there is associated (sequence-like) state in supported DBs that needs to be managed in functions like `sequence_reset_sql`. I haven't replaced other `isinstance(..., AutoField)` calls for this reason, though there may be some others worth changing.